http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpWsmDao.scala

```diff
@@ -125,7 +129,12 @@ class HttpWsmDao[F[_]](httpClient: Client[F], config: HttpWsmDaoConfig)(implicit
       )(onError)
     } yield res
 
-  override def getLandingZoneResources(billingProfileId: String, userToken: Authorization)(implicit
+  override def getLandingZoneResources(billingProfileId: BillingProfileId, userToken: Authorization)(implicit
+    ev: Ask[F, AppContext]
+  ): F[LandingZoneResources] =
+    lzResourcesCache.cachingF(billingProfileId)(None)(getLandingZoneResourcesInternal(billingProfileId, userToken))
```

Comment:
Added a cache in front of the LZ resources query, since it's an expensive call and LZs are immutable.

Reply:
Just a thought, but I am wondering if it would be worth putting this as a comment into the code, it would help a lot I think

Reply:
Is the cache bounded in any way or do items age out?

I am also concerned we are adding complexity to overfit for the workshop use case of lots of concurrent users in a single landing zone. Alternatively could we just make the call to get LZ info cheaper?

Reply:
Is the cache bounded in any way or do items age out?

I am also concerned we are adding complexity to overfit for the workshop use case of lots of concurrent users in a single landing zone. Alternatively could we just make the call to get LZ info cheaper?

Reply:
The cache size/expiry is defined in Boot.scala with numbers I picked out of thin air: https://github.com/DataBiosphere/leonardo/pull/3814/files#diff-8af5da8d5edb93397a2e5c102c53047173dac803b46343c12568196a90cee553R435-R438

```
underlyingLandingZoneCache = buildCache[BillingProfileId, scalacache.Entry[LandingZoneResources]](
        500,
        4 hours
      )
```

I don't think it's a huge cost in complexity, we use the same caching pattern for several things in Leo.

The one downside I was thinking about is if there's a destructive LZ upgrade which removes/renames things, making the cached value incorrect. But I hope we don't do those kinds of LZ updates in general -- they should be additive/backwards compatible.

Reply:
I don't actually care about the cache _that_ much -- the main thing was moving it out of the path of the user request.

Reply:
I have lost count of all my caching related scars. If you don't care that much about the caching, I would remove it and use this as an excuse to make the LZ api better.